src/middleware/gifFetcher.js

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+import trendingGifsActions from '../actions/trendingGifs';
+
+const { TRENDING_GIFS_GET } = trendingGifsActions.types;
+const { SEARCH_GIFS_GET } = trendingGifsActions.types;
+const { ADD_FAV } = trendingGifsActions.types;
+const { REMOVE_FAV } = trendingGifsActions.types;
+const { fetchedTrendingGifs } = trendingGifsActions.creators;
+const { fetchedSearchGifs } = trendingGifsActions.creators;
+
+const API_URL = 'https://api.giphy.com/v1/gifs/';
+
+const fetchTrending = (dispatch) => {
+  axios.get(`${API_URL}trending?api_key=SA3WrTOcAq5bNF9hsWC4Z8j0O1UkOMuB&limit=10&rating=G`)
```

Comment:
I'll advise that you put your api_key in .env and provide a .env.example

Reply:
https://www.npmjs.com/package/dotenv